### ISSUE TITLE
iPad Support?

### ISSUE THREAD

**Issue by Macley-Kun** [midnightchip/Asteroid#14]
So i got this cool looking tweak for my iPad 7,5 11.4.1. 

I do know that the weather.app isn't a thing for iPad's. But i was wondering if there would be support for iPad's.

Installing and configuring works btw perfectly, love the aesthetics and details that are put in into this tweak. Like the respring for example.

I only got as far that i enabled just the lockscreen settings. Then i kept being stuck on the respring part (about 80% done).

**Comment by the-casle** [comment]
Thats a great idea. Can you check in settings if theres any location permissions for weather? Also, try installing the version from my repo:
https://the-casle.github.io/cydia

**Comment by Macley-Kun** [comment]
So I wasn’t to sure if I had your version, but now I have. Now the following happens:
Installing the tweak (even with the lib) makes the device stuck in respring. When I disable all other tweaks I get into safe mode...

And in settings there aren’t any permissions for weather, the only weather settings that can be used is from third party. Like Yahoo Weer! (Weather).

Why the tweak suddenly doesn’t work at all on my iPad is not clear to me. The only thing I could blame is that iPads don’t have a default weather app like iPhones have.

Thank you for considering this suggestion!
All tweaks I have currently installed: 
https://hastebin.com/asuyuqocox.makefile

**Comment by Macley-Kun** [comment]
So funny update on my side.
From https://github.com/JunesiPhone/XenInfo/issues/7 i found out that i didn't use the native weather for over a year..

I now have Astroid installed and hitting enable doesn't crash the device! I haven't played with the settings yet, but will do in the comming days (or maybe today).

**Comment by Macley-Kun** [comment]
macley.stackstorage.com/s/VqXZVfEwhhEIttnH
macley.stackstorage.com/s/fNdFCJ0aJACKXScp
macley.stackstorage.com/s/tLF1MRBk11WEotqk
macley.stackstorage.com/s/X1DAxiGLieXIcHDE
Above are the biggest bugs I’ve found, these are common due to iPads being able to rotate. I think colorflow 4 has a good implementation to handle this issue.

Please feel free to ask any questions :)

**Comment by Macley-Kun** [comment]
Updated to ios 12.1.1, think Asteroid first needs to be supported before we can look back again on the alligment.

**Comment by the-casle** [comment]
https://the-casle.github.io/cydia/

That repo has a work in progress iOS 12 version. It also doesn’t depend on libcsweather or csweather. I’m working on my other tweak nine, but also getting around to Asteroid.

**Comment by Macley-Kun** [comment]
np dude, take ya time!
the tweak installs nicely, background (live) works (not proper for ipad but that's excpected) don't see any hello/message but maybe i should wait a bit longer until tommorow or so 📃

**Comment by the-casle** [comment]
It should just show up when theres no notifications on the lockscreen. Might be a bug, also how off is the the live weather?

**Comment by Macley-Kun** [comment]
Um, i have no information whatsoever about live weather... Like i have the following ON:
Enable ON
Refresh: 30 seconds
Lockscreen greeting view: ON
The live weather does work, it's just aspect ratio issue what is present (same pictures as i showed earlier).
v1.3.5. Xeninfo does work fyi!

**Comment by the-casle** [comment]
I’m not sure where xenhtml gets its weather, but it’s open source. Thanks for info.

**Comment by Macley-Kun** [comment]
xenhtml widgets normally use [xeninfo](http://junesiphone.com/xeninfo/). The latest version does support IOS 12 and the weather works on iPads too!

I don't want to say i know anything about programming/libraries/tweakdevelopers, but i do assume the following why recieving weather works sometimes and sometimes not for iPads in particular.

iPads don't have the weather app, therefore they cannot set anything else besides the default configuration.
The default configuration, is only one tab, with the current location information (you can't remove this one). Some tweaks have difficulties recieving this information, where they provide a workaround to add a second static location.
(This is also sometimes the reason to enable weather to always)

Here is a small list of changes in my opinion that should be implemented to have proper iPad support:

- Aspect ratio for live weather lock & home screen: https://macley.stackstorage.com/s/fNdFCJ0aJACKXScp & https://macley.stackstorage.com/s/tLF1MRBk11WEotqk
- Being able to use the default weather configuration (live location) to display the weather information.
- Aspect ratio of the forecast for the lockscreen when it's initilized to use the portrait mode https://macley.stackstorage.com/s/tLF1MRBk11WEotqk 
- Moving the dismiss button slightly more down. Adding a bit more spacing from the left and right side: https://macley.stackstorage.com/s/VqXZVfEwhhEIttnH
- Have the greeting moment and the forecast not happen at the same time: https://macley.stackstorage.com/s/VqXZVfEwhhEIttnH
- Removing the option to configure the weather app's look, as there is none on the iPad

Sorry if it looks a bit confusing, but i really try to be as precise as possible as debugging this withoud an actual device in your hands can be a bit troublesum.
~~i don't have an iPad mini or Pro or 10inch device, but i assume we keep those out of the scope~~

**Comment by the-casle** [comment]
Greeting view and forecast can be selected as on or off in the lockscreen components section

That’s actually a really useful bunch of links and comments. Thank you!

**Comment by Macley-Kun** [comment]
Yeah j have that option on, and the live weather does kinda work, but I no longer see the greeting view, with or without other tweaks strangely enough.
Could maybe try remove the plist file and tweak and retry again :thinking:

Outlook voor iOS<https://aka.ms/o0ukef> downloaden

________________________________
Van: the-casle <notifications@github.com>
Verzonden: dinsdag, maart 5, 2019 7:24 PM
Aan: midnightchip/Asteroid
CC: Macley; Author
Onderwerp: Re: [midnightchip/Asteroid] iPad Support? (#14)


Greeting view and forecast can be selected as on or off in the lockscreen components section

That’s actually a really useful bunch of links and comments. Thank you!

―

**Comment by Macley-Kun** [comment]
Small update, I was stupid enough to not go into components and actually enable the weather stuff that I previously saw. So showing any weather info works, so I believe it’s just lay-out issues.

With that said. I noticed this issue #23 what describes a similar issues as on the iPad.
I think the main focus should be brought to the live BG view first, then how components for both views.

**Comment by Macley-Kun** [comment]
Also, i believe the background animations aren't loaded correctly at all, i saw a video on yt the other day and was suprised how amazing it looked.
Previous pictures also show it's just a blue even background, may i ask where exactly these animations like thunderclouds are pulled from?

**Comment by the-casle** [comment]
Hi, sorry I haven’t responded in a bit. Had some issues compiling. The animations are under the weather.framework or the weatherui.framework. 

I also got the rotation figured out, just the basics however. It would be super helpful to check if those frameworks exist, and if they do send as a zip so I can make sure its all there.

**Comment by Macley-Kun** [comment]
I found one of the frameworks, weather.framework. The UI version isn't sadly present (expected as there is no weather app 🙄).

I zipped you the one i found, this is the path it is in.
[weather.framework.zip](https://github.com/midnightchip/Asteroid/files/3064803/weather.framework.zip)
![IMG_0101](https://user-images.githubusercontent.com/26381427/55895982-93e95600-5bbd-11e9-9f5e-2e54641922f7.PNG)

Aah ic, makes sence where you got them from, i don't know if it's possible for iPad's (or weather UI less devices) to use some fallback animations, but that's up to you if that's something to try out.

Ah thanks for the compliment ^^, i really love how the app looks and when A12 get's jailbroken it'll be the second tweak i'll install (first a backup tweak hehe).

My Discord is Macley#9105, i'm ok sharing this online :P
